src/OmniSharp.Roslyn.CSharp/CodeActions/CodeActionHelper.cs

```diff
@@ -1,102 +1,32 @@
-
-            // Try to find exact match
-            var pointDiagnostics = diagnostics.Where(d => d.Location.SourceSpan.Equals(span)).ToImmutableArray();
-            // No exact match found, try approximate match instead
-            if (pointDiagnostics.Length == 0)
-            {
-                var firstMatchingDiagnostic = diagnostics.FirstOrDefault(d => d.Location.SourceSpan.Contains(span));
-                // Try to find other matches with the same exact span as the first approximate match
-                if (firstMatchingDiagnostic != null)
-                {
-                    pointDiagnostics = diagnostics.Where(d => d.Location.SourceSpan.Equals(firstMatchingDiagnostic.Location.SourceSpan)).ToImmutableArray();
-                }
-            }
-            // At this point all pointDiagnostics guaranteed to have the same span
-            if (pointDiagnostics.Length > 0)
-            {
-                return new CodeFixContext(originalDocument, pointDiagnostics[0].Location.SourceSpan, pointDiagnostics, (a, d) => actionsDestination.Add(a), CancellationToken.None);
-            }
-
-            return null;
-        }
-
-        private static TextSpan GetTextSpan(ICodeActionRequest request, SourceText sourceText)
-        {
-            if (request.Selection != null)
-            {
-                var startPosition = sourceText.Lines.GetPosition(new LinePosition(request.Selection.Start.Line, request.Selection.Start.Column));
-                var endPosition = sourceText.Lines.GetPosition(new LinePosition(request.Selection.End.Line, request.Selection.End.Column));
-                return TextSpan.FromBounds(startPosition, endPosition);
-            }
-            var position = sourceText.Lines.GetPosition(new LinePosition(request.Line, request.Column));
-            return new TextSpan(position, 1);
-        }
-
-        private static readonly HashSet<string> _blacklist = new HashSet<string> {
-            // This list is horrible but will be temporary
-            "Microsoft.CodeAnalysis.CSharp.CodeFixes.AddMissingReference.AddMissingReferenceCodeFixProvider",
+            "Microsoft.CodeAnalysis.CSharp.AddMissingReference.CSharpAddMissingReferenceCodeFixProvider",
             "Microsoft.CodeAnalysis.CSharp.CodeFixes.Async.CSharpConvertToAsyncMethodCodeFixProvider",
             "Microsoft.CodeAnalysis.CSharp.CodeFixes.Iterator.CSharpChangeToIEnumerableCodeFixProvider",
-            "Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ChangeSignature.ChangeSignatureCodeRefactoringProvider",
+            "Microsoft.CodeAnalysis.ChangeSignature.ChangeSignatureCodeRefactoringProvider",
+            "Microsoft.CodeAnalysis.ExtractInterface.ExtractInterfaceCodeRefactoringProvider"
+        };
+
+        private static readonly HashSet<string> _nrefactoryListToRemove = new HashSet<string>
+        {
             "ICSharpCode.NRefactory6.CSharp.Refactoring.CreateMethodDeclarationAction",
```

Comment:
Can probably just drop this list for now (perhaps even the hashset).

Currently Refactoring Essentials isn't built with the latest roslyn dlls, so the plugins branch fails.  I'll have to get some other form of unit test in there to ensure that plugins are working.